crates/router/src/core/payments/access_token.rs

```diff
@@ -196,3 +271,150 @@ pub async fn refresh_connector_auth(
+        connector_meta_data: None,
+        amount_captured: None,
+        access_token: None,
+        session_token: None,
+        reference_id: None,
+        request: access_token_request,
+        response: Err(types::ErrorResponse::default()),
+        payment_method_id: None,
+    })
+}
+
+pub async fn get_access_token_retry_schedule_time(
+    db: &dyn db::StorageInterface,
+    connector: &str,
+    merchant_id: &str,
+    retry_count: i32,
+) -> Result<Option<time::PrimitiveDateTime>, errors::ProcessTrackerError> {
+    let pt_mapping_key = &format!("pt_mapping_refund_sync_{connector}");
+    let custom_merchant_mapping = db.find_config_by_key(pt_mapping_key).await?;
+
+    let connector_mapping = custom_merchant_mapping
+        .config
+        .parse_struct::<ConnectorPTMapping>("CustomPTMapping");
+
+    let mapping = match connector_mapping {
+        Ok(x) => x,
+        Err(err) => {
+            crate::logger::error!("Error: while getting connector mapping: {}", err);
+            process_data::ConnectorPTMapping::default()
+        }
+    };
+
+    let time_delta = scheduler_utils::get_schedule_time(mapping, merchant_id, retry_count + 1);
+
+    Ok(scheduler_utils::get_time_from_delta(time_delta))
+}
+
+#[async_trait::async_trait]
+impl ProcessTrackerWorkflow for AccessTokenRefresh {
+    async fn execute_workflow<'a>(
+        &'a self,
+        state: &'a AppState,
+        process: storage::ProcessTracker,
+    ) -> Result<(), errors::ProcessTrackerError> {
+        let tracking_data: ProcessTrackerAccessTokenData = process
+            .tracking_data
+            .clone()
+            .parse_value("AccessTokenData")?;
+        let db: &dyn db::StorageInterface = &*state.store;
+
+        let merchant_connector_account = db
+            .find_merchant_connector_account_by_merchant_id_connector(
+                &tracking_data.merchant_id,
+                &tracking_data.connector,
+            )
+            .await?;
+
+        let access_token_router_data =
+            construct_access_token_router_data(&tracking_data, &merchant_connector_account)?;
+
+        let connector = types::api::ConnectorData::get_connector_by_name(
+            &state.conf.connectors,
+            &tracking_data.connector,
+            types::api::GetToken::Connector,
+        )?;
+
+        let new_access_token_result =
+            refresh_connector_auth(state, &connector, &access_token_router_data).await?;
+
+        match new_access_token_result {
+            Ok(access_token) => {
+                // Update the access token in redis
+                let store = &*state.store;
+                let _ = state
+                    .store
+                    .set_access_token(
+                        &tracking_data.merchant_id,
+                        connector.connector.id(),
+                        access_token.clone(),
+                    )
+                    .await
```

Comment:
Can you add comments over here on why the error should not be propagated.

Reply:
This flow will be on the critical path of `payments`. Once we have the access token, if saving it in the database (redis) fails, we should not fail the payments. The next payment will again create new access token.

Reply:
I have drafted this PR as of now, as there are some changes that have been merged into the main which will have conflicts with this change. Will pick this up later. I will add comments.

Reply:
Sure. Can you add the same comments in the codebase as well. Also if errors should not be thrown then why is there a fallible function on line 395.